src/integration/java/io/pinecone/helpers/IndexManagerSingleton.java

```diff
@@ -0,0 +1,208 @@
+package io.pinecone.helpers;
+
+import io.pinecone.clients.Index;
+import io.pinecone.clients.Pinecone;
+import io.pinecone.configs.PineconeConfig;
+import io.pinecone.exceptions.PineconeException;
+import io.pinecone.proto.DescribeIndexStatsResponse;
+import org.openapitools.client.model.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static io.pinecone.helpers.BuildUpsertRequest.buildRequiredUpsertRequestByDimension;
+import static io.pinecone.helpers.IndexManager.*;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class IndexManagerSingleton {
```

Comment:
This is the most crucial part of this work. The goal here is to have a centralized singleton that manages all the index resources across integration tests explicitly, and is able to tear things down at the end.

Reply:
I think we should consider renaming this class to make it clear that it's a test utility, not part of the client itself.

Naming alternatives?
* `TestResourceManager`
* `TestIndexResourcesManager`
* `IndexTestingResourcesManager`